file-system.ts

```diff
@@ -443,5 +444,28 @@ export class FileSystem implements IFileSystem {
 		}
 		return foundFiles;
 	}
+
+	public getFileShasum(fileName: string, encoding?: string): IFuture<string> {
+		let future = new Future<string>();
+		encoding = encoding || "sha1";
+		let logger: ILogger = this.$injector.resolve("logger");
```

Comment:
I prefer to name variables from injector with leading $, so that searching manually with text editor finds them more easily.